server/src/test/java/org/elasticsearch/index/mapper/DynamicTemplatesTests.java

```diff
@@ -1383,4 +1383,376 @@ public void testSubobjectsFalseWithInnerNestedFromDynamicTemplate() {
             exception.getRootCause().getMessage()
         );
     }
+
+    public void testDynamicSubobject() throws IOException {
+        MapperService mapperService = createMapperService(topMapping(b -> {
```

Comment:
Is it worth building these mappings as a text block as well? I think it reads much more easily than xcontentbuilders

Reply:
yes, the tricky bit is that we define only part of the mappings, and part is defined in topMapping. I would leave this for another time to be honest :)

Reply:
`topMapping` is just `{"_doc":{ ... } }` though.  Up to you :)

Reply:
I'd prefer to spend time on fixing dynamic object mappers lookup in the context etc. over rewriting tests that work fine :)